aws/rust-runtime/aws-sig-auth/src/signer.rs

```diff
@@ -146,38 +145,45 @@ impl SigV4Signer {
         } else {
             PayloadChecksumKind::NoHeader
         };
-        let sigv4_config = aws_sigv4::http_request::SigningParams {
-            access_key: credentials.access_key_id(),
-            secret_key: credentials.secret_access_key(),
-            security_token: credentials.session_token(),
-            region: request_config.region.as_ref(),
-            service_name: request_config.service.as_ref(),
-            date_time: request_config.request_ts.into(),
-            settings,
+        let sigv4_config = {
+            let mut builder = SigningParams::builder()
+                .access_key(credentials.access_key_id())
+                .secret_key(credentials.secret_access_key())
+                .region(request_config.region.as_ref())
+                .service_name(request_config.service.as_ref())
+                .date_time(request_config.request_ts.into())
+                .settings(settings);
+            builder.set_security_token(credentials.session_token());
+            builder.build().unwrap()
```

Comment:
nit: probably should `expect("all fields set")` or similar. (Or just use the private struct constructor since we're inside the crate)